plugins/wasm-rust/src/redis_wrapper.rs

```diff
@@ -0,0 +1,661 @@
+use std::{collections::HashMap, time::Duration};
+
+use proxy_wasm::{hostcalls::RedisCallbackFn, types::Status};
+use redis::{Cmd, ToRedisArgs, Value};
+
+use crate::{cluster_wrapper::Cluster, internal};
+
+pub type RedisValueCallbackFn = dyn FnOnce(&Result<Value, String>, usize, u32);
+
+fn gen_callback(call_fn: Box<RedisValueCallbackFn>) -> Box<RedisCallbackFn> {
+    Box::new(move |token_id, status, response_size| {
+        let res = match internal::get_redis_call_response(0, response_size) {
+            Some(data) => match redis::parse_redis_value(&data) {
+                Ok(v) => Ok(v),
+                Err(e) => Err(e.to_string()),
+            },
+            None => Err("response data not found".to_string()),
+        };
+        call_fn(&res, status, token_id);
+    })
+}
+pub struct RedisClientBuilder {
```

Comment:
as above, well formated should keep newlines between different blocks

Reply:
done